File: backend/internal/hrm/checkin/model/employee_workshift.go
package model

import (
	"time"
)

type EmployeeWorkshift struct {
	ID          uint      `gorm:"primaryKey;autoIncrement" json:"id"`
	EmployeeID  string    `gorm:"column:employee_id;index:idx_employee_date" json:"employee_id"`
	WorkShiftID string    `gorm:"column:workshift_id;type:varchar" json:"workshift_id"`
	Date        time.Time `gorm:"column:date;index:idx_employee_date;type:timestamp" json:"date"`
	CreatedAt   time.Time `gorm:"column:created_at;" json:"created_at"`

	WorkShift WorkShifts `gorm:"foreignKey:WorkShiftID;references:WorkShiftID;constraint:OnUpdate:CASCADE,OnDelete:RESTRICT" json:"workshift"`
}

func (EmployeeWorkshift) TableName() string {
	return "employee_workshift"
}

File: frontend/src/types/employee-workshift.ts
import type {Workshift} from "@/types/workshift.ts";

export type EmployeeWorkshift = {
    id: string;
    CreatedAt: string;
    UpdatedAt: string;
    DeletedAt: string;
    employee_id: string;
    workshift_id: string;
    workshift: Workshift
}

export interface EmployeeWorkshiftResponse {
    [employeeId: string]: EmployeeWorkshift[];
}

export interface RegisterWorkshiftRequest {
    employee_id: string;
    workshift_id: string;
    date: string;
}